src/engraving/infrastructure/io/mscreader.cpp

```diff
@@ -244,24 +246,29 @@ MscReader::ZipFileReader::~ZipFileReader()
     }
 }
 
-bool MscReader::ZipFileReader::open(IODevice* device, const path_t& filePath)
+Ret MscReader::ZipFileReader::open(IODevice* device, const path_t& filePath)
 {
     m_device = device;
     if (!m_device) {
         m_device = new File(filePath);
         m_selfDeviceOwner = true;
     }
 
+    if (!FileInfo::exists(filePath)) {
+        LOGD() << "not exists path: " << filePath;
+        return scoreFileErrorToRet(Score::FileError::FILE_NOT_FOUND, filePath);

```

Comment:
this is wrong in the case that the filePath is just a dummy name (for logging and whatever other purposes), but the actual data passed in is via the "device" argument, which is why one of the unit tests is failing

Reply:
What is the preferred solution to this?

Reply:
If you move this inside the !m_device check it might work.  Otherwise to the calling function.